src/main/java/com/mparticle/kits/GoogleAnalyticsFirebaseKit.java

```diff
@@ -128,7 +132,22 @@ public String getName() {
                 eventName = FirebaseAnalytics.Event.SELECT_CONTENT;
                 break;
             case Product.CHECKOUT_OPTION:
-                eventName = FirebaseAnalytics.Event.SET_CHECKOUT_OPTION;
+                Map<String, List<String>> customFlags = commerceEvent.getCustomFlags();
+                if (customFlags != null && customFlags.containsKey(CF_GA4COMMERCE_EVENT_TYPE)) {
+                    String commerceEventType = customFlags.get(CF_GA4COMMERCE_EVENT_TYPE).get(0);
```

Comment:
this could still throw an IndexOutOfBounds exception if the array is empty (if for some reason the client set `event.addCustomFlag(GF_GA4CommerceEventType, new ArrayList<String>())`

it's annoying and really verbose to do this in Java, but the safe way to pull out these custom flag would be

```
Map<String, List<String>> customFlags = commerceEvent.getCustomFlags();
                if (customFlags != null && customFlags.containsKey(CF_GA4COMMERCE_EVENT_TYPE)) {
                    List<String> commerceEventTypes = customFlags.get(CF_GA4COMMERCE_EVENT_TYPE);
                    if (!commerceEventTypes.isEmpty()) {
                        String commerceEventType = commerceEventTypes.get(0);
```

Reply:
@willpassidomo i see.  good for another look.  i then realized i have some code here that is similar, but uses .size() > 0...should be no issues keeping that right, or do you have any adjustments to make there as well?https://github.com/mparticle-integrations/mparticle-android-integration-google-analytics-firebase/pull/37/files#diff-33c0739ab1e9a09e8697643625184bfaf8803b4e8edd6fe9eb58141c6f19e243R243-R254